frame/timestamp/src/lib.rs

```diff
@@ -164,7 +164,7 @@ decl_module! {
 		/// - 1 event handler `on_timestamp_set` `O(T)`.
 		/// # </weight>
 		#[weight = (
-			T::WeightInfo::set(),
+			T::WeightInfo::set(UniqueSaturatedInto::<u32>::unique_saturated_into(*now)),
```

Comment:
This is the only fishy change, but I don't see another way for the moment.

Reply:
No - just make `on_timestamp_set` be required to be O(1).